### PR TITLE
feat: introduce http service to fetch resources with proxy and certificate authorities for https configured

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Provides http service allowing to fetch resources by having proxy being configured and certificate authority configured as well (in case of https resource)

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-1702

### How to test this PR?
Use this service or look at tests on how it should fetch resources
There will be a follow-up with recommendation plug-in using this service

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

**Comment by benoitf** [comment]
[crw-ci-test --rebuild]

File: extensions/eclipse-che-theia-plugin-ext/src/browser/che-http-main.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import { CheHttpMain } from '../common/che-protocol';
+import { HttpService } from '@eclipse-che/theia-remote-api/lib/common/http-service';
+import { interfaces } from 'inversify';
+
+export class CheHttpMainImpl implements CheHttpMain {
+  private readonly cheHttpervice: HttpService;
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-plugin-ext/src/browser/che-http-main.ts:16]: cheHttp**S**ervice

File: extensions/eclipse-che-theia-remote-impl-che-server/src/node/che-server-http-service-impl.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      throw error;
+    }
+  }
+
+  /**
+   * Use proxy and/or certificates.
+   */
+  private async getAxiosInstance(uri: string): Promise<AxiosInstance> {
+    const certificateAuthority = await this.certificateService.getCertificateAuthority();
+
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-remote-impl-che-server/src/node/che-server-http-service-impl.ts:48]: we definitely need some shared component for that
https://github.com/eclipse-che/che-theia/blob/master/extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts#L129
maybe not in this PR

Comment (benoitf) [reply extensions/eclipse-che-theia-remote-impl-che-server/src/node/che-server-http-service-impl.ts:48]: yes I want to migrate to this one after

**Review by azatsarynnyy — APPROVED** [review]

**Comment by benoitf** [comment]
happy paths passed on github actions, merging

**Pull request merged**